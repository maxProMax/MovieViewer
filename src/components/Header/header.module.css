.container {
    position: sticky;
    top: 0;
    background: white;
    z-index: 2;
    box-shadow: 0 4px 2px -2px rgb(0 0 0 / 20%);
}

.wrapper {
    max-width: 1280px;
    margin: auto;
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 48px;
    padding: 0 24px;
}

.logo {
    font-weight: 700;
    text-shadow: 2px 1px #c2c2c2;
}

.burger {
    display: inline-flex;
    background: none;
    border: none;
    padding: 0;
}
