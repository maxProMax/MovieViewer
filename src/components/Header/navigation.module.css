.nav {
    padding: 48px 24px 24px;
    width: 200px;
}

.list {
    display: flex;
    flex-direction: column;
    row-gap: 24px;
}

.item {
    cursor: pointer;
}
