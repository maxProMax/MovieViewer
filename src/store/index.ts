export { store } from './store';
export { useInitRedux } from './hooks';
