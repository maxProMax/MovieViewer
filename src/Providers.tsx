import { FC, PropsWithChildren } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { store } from './store';

export const GlobalProvider: FC<PropsWithChildren> = ({ children }) => {
    return (
        <BrowserRouter>
            <Provider store={store}>{children}</Provider>
        </BrowserRouter>
    );
};
