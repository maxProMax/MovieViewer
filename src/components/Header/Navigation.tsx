import { FC, PropsWithChildren } from 'react';
import { List } from 'src/components/common/List';
import { LinkFavorites, LinkWatchLater, LinkHome } from 'src/router/links';
import { useFavorites, useWatchLatter } from 'src/store/hooks';
import styles from './navigation.module.css';

const Li: FC<PropsWithChildren> = ({ children }) => (
    <li className={styles.item}>{children}</li>
);

export const Navigation: FC = () => {
    const { count: favorites } = useFavorites();
    const { count: watchLater } = useWatchLatter();

    return (
        <nav className={styles.nav}>
            <List className={styles.list}>
                <Li>
                    <LinkHome>Home</LinkHome>
                </Li>
                <Li>
                    <LinkFavorites>
                        Favorites {favorites ? `(${favorites})` : ''}
                    </LinkFavorites>
                </Li>
                <Li>
                    <LinkWatchLater>
                        Watch Later {watchLater ? `(${watchLater})` : ''}
                    </LinkWatchLater>
                </Li>
            </List>
        </nav>
    );
};
