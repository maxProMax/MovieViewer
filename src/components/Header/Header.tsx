import { FC, useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';
import MenuIcon from '@mui/icons-material/Menu';
import Drawer from '@mui/material/Drawer';
import { LinkHome } from 'src/router/links';
import { Navigation } from './Navigation';
import styles from './header.module.css';

export const Header: FC = () => {
    const [open, change] = useState(false);
    const location = useLocation();

    useEffect(() => {
        change(false);
    }, [location]);

    return (
        <header className={styles.container}>
            <div className={styles.wrapper}>
                <button className={styles.burger} onClick={() => change(true)}>
                    <MenuIcon />
                </button>
                <Drawer open={open} onClose={() => change(false)}>
                    <Navigation />
                </Drawer>
                <div className={styles.logo}>
                    <LinkHome>MovieViewer</LinkHome>
                </div>
            </div>
        </header>
    );
};
