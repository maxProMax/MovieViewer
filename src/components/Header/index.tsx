import { FC, useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';
import MenuIcon from '@mui/icons-material/Menu';
import Drawer from '@mui/material/Drawer';
import { Navigation } from './Navigation';
import { LinkHome } from '../../router/links';
import styles from './header.module.css';

export const Header: FC = () => {
    const [open, change] = useState(false);
    const location = useLocation();

    useEffect(() => {
        change(false);
    }, [location]);

    return (
        <header className={styles.container}>
            <div className={styles.wrapper}>
                <span className={styles.burger} onClick={() => change(true)}>
                    <MenuIcon />
                </span>
                <Drawer open={open} onClose={() => change(false)}>
                    <Navigation />
                </Drawer>
                <div className={styles.logo}>
                    <LinkHome>MovieViewer</LinkHome>
                </div>
            </div>
        </header>
    );
};
