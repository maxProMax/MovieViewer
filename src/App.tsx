import { FC } from 'react';
import { Body } from './components/Body';
import { Footer } from './components/Footer';
import { Header } from './components/Header';
import { useInitRedux } from './store';
import './App.css';

export const App: FC = () => {
    useInitRedux();

    return (
        <>
            <Header />
            <Body />
            <Footer />
        </>
    );
};

export default App;
