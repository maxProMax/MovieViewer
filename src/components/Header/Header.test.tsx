import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { GlobalProvider } from 'src/Providers';
import { Header } from './Header';

describe('Header', () => {
    test('open sidebar', async () => {
        render(<Header />, { wrapper: GlobalProvider });

        userEvent.click(screen.getByRole('button'));

        ['Favorites', 'Home', 'Watch Later'].forEach((value) => {
            const el = screen.getByText(new RegExp(value, 'i'));

            expect(el).toBeInstanceOf(HTMLAnchorElement);
            expect(el).toBeInTheDocument();
        });
    });

    test('Layout', () => {
        const view = render(<Header />, { wrapper: GlobalProvider });

        expect(view.container).toMatchSnapshot();
    });
});
