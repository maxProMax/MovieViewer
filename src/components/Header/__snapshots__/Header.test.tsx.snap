// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Layout 1`] = `
<div>
  <header
    class="container"
  >
    <div
      class="wrapper"
    >
      <button
        class="burger"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        class="logo"
      >
        <a
          aria-current="page"
          class="link active"
          href="/"
        >
          MovieViewer
        </a>
      </div>
    </div>
  </header>
</div>
`;
